### PR TITLE
feat: dwh event fallback and proper contexts

### PR DESCRIPTION
**Pull request by oopcode**
* retry for failed events
* proper context handling

### CODE REVIEW

File: cmd/dwh/main.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func run() {
 
 	w, err := dwh.NewDWH(ctx, cfg, key)
 	if err != nil {
-		log.G(ctx).Error("cannot start DWH service", zap.Error(err))
+		log.G(ctx).Error("cannot create NewDWH", zap.Error(err))
```
Comment (antmat) [inline cmd/dwh/main.go]: cannot create NewDWH ?

File: cmd/dwh/main.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ func run() {
 	log.G(ctx).Info("starting DWH service", zap.String("grpc_addr", cfg.GRPCListenAddr),
 		zap.String("http_addr", cfg.HTTPListenAddr))
 	if err := w.Serve(); err != nil {
-		log.G(ctx).Error("cannot start DWH service", zap.Error(err))
+		log.G(ctx).Error("DWH failed", zap.Error(err))
```
Comment (antmat) [inline cmd/dwh/main.go]: It will be smth like "DWH failed - context canceled" during shutdown, won't it?

File: insonmnia/dwh/server.go [hunk 3]
```diff
@@ -820,26 +817,55 @@ func (w *DWH) watchMarketEvents() error {
...
-	}
+			// Events in the same block can come in arbitrary order. If two events have to be processed
+			// in a specific order (e.g., OrderPlaced > DealOpened), we need to retry if the order is
+			// messed up.
+			if err := w.processEvent(event); err != nil {
+				w.logger.Error("failed to processEvent, retrying", zap.Error(err),
+					zap.Uint64("block_number", event.BlockNumber),
+					zap.String("event_type", reflect.TypeOf(event.Data).Name()))
+
+				go func() {
```
Comment (antmat) [inline insonmnia/dwh/server.go]: better extract method here

File: cmd/dwh/main.go (hunk 2 above)
Comment (oopcode) [reply cmd/dwh/main.go]: "context cancelled" is already present in the error itself:

```
// Canceled is the error returned by Context.Err when the context is canceled.
var Canceled = errors.New("context canceled")
```

File: cmd/dwh/main.go (hunk 1 above)
Comment (oopcode) [reply cmd/dwh/main.go]: you mean, missing space?

**Pull request merged**